### DISCUSSION TITLE
Smartshape2D terrain

### DISCUSSION THREAD

**Discussion by ActionGameMaster** [Techokami/SonicWorldsNext#121] [Q&A]
I want to try and use smartshape2d to make terrain, slopes and loops in the engine, how do I add collision to it so the characters don't go through it?

**Comment by Renhoex** [comment]
Take a look at the collission masks, for the player to interact the first 4 bits should be set 
your collision mask and layer should look like this, if you're not sure what these are remember to hover over the label names.
![image](https://github.com/Techokami/SonicWorldsNext/assets/26295900/ddc5e87d-65cb-4086-b244-0670aee37e58)

If you're trying to create collisions I'd suggest reading the smartshape2D collision guides since that's not really related to worlds next

**Reply by ActionGameMaster** [reply]
how do I make walls from the shapes?

**Reply by Renhoex** [reply]
again, go to the static body smart shapes generates and set up your layer masks like I did in the image.

**Reply by ActionGameMaster** [reply]
I got it, now how do I do loop de-loops with them?

**Reply by ActionGameMaster** [reply]
nvmnd, i figured out the high and low layer masks
